docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+---
+title: Unit testing in Office Add-ins
+description: 'Learn how to unit test code that calls the Office JavaScript APIs'
+ms.date: 11/14/2021
+ms.localizationpriority: medium
+---
+
+# Unit testing in Office Add-ins
+
+It is a good practice to create unit tests for your add-in's code. Unit testing server-side code, and client-side code that does *not* call the [Office JavaScript APIs](../develop/understanding-the-javascript-api-for-office.md), is the same in Office Add-ins as it is in any web application, so it requires no special documentation. But client-side code that calls the Office JavaScript APIs is challenging to test for the following reasons:
+
+- The Office JavaScript APIs must initialize in a webview control in the context of an Office application (Excel, Word, etc.), so they cannot be loaded in the process in which unit tests run on your development computer.
+- Some unit testing frameworks require that mock objects emulate all the members of the object type that is being mocked, which isn't feasible for the Office JavaScript APIs because some object types have over a hundred members.
+- Unit testing frameworks are stateless, which makes it difficult to test the [application-specific APIs](../develop/understanding-the-javascript-api-for-office.md#api-models) in the Office JavaScript APIs, because they have methods [load()](../develop/application-specific-api-model.md#load) and [sync()](../develop/application-specific-api-model.md#sync) that must be called in a particular order relative to other functions and to each other.
+
+To solve these problems, we have created a library to simplify the creation of mock Office objects in unit tests: [Office-Addin-Mock](https://github.com/OfficeDev/Office-Addin-Scripts/tree/master/packages/office-addin-mock).
```

Comment:
I think it is better for this link to redirect to the [Npm page](https://www.npmjs.com/package/office-addin-mock), as I think it is cleaner and easier to see how to install right away.